packages/components/src/components/modal/modal.tsx

```diff
@@ -175,61 +206,66 @@ const ModalElement = ({
     );
 };
 
-ModalElement.defaultProps = {
-    has_close_icon: true,
-    should_header_stick_body: true,
-};
-
-ModalElement.propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-    close_icon_color: PropTypes.string,
-    has_close_icon: PropTypes.bool,
-    header_backgound_color: PropTypes.string,
-    header: PropTypes.node,
-    id: PropTypes.string,
-    is_open: PropTypes.bool,
-    is_title_centered: PropTypes.bool,
-    onMount: PropTypes.func,
-    onUnmount: PropTypes.func,
-    small: PropTypes.bool,
-    should_header_stick_body: PropTypes.bool,
-    renderTitle: PropTypes.func,
-    title: PropTypes.oneOfType([PropTypes.string, PropTypes.bool, PropTypes.node]),
-    toggleModal: PropTypes.func,
-    elements_to_ignore: PropTypes.array,
+type TModal = {
+    className?: string;
+    close_icon_color?: string;
+    elements_to_ignore?: HTMLElement[];
+    exit_classname?: string;
+    has_close_icon?: boolean;
+    header?: React.ReactNode;
+    header_background_color?: string;
+    height?: string;
+    id?: string;
+    is_confirmation_modal?: boolean;
+    is_open: boolean;
+    is_risk_warning_visible?: boolean;
+    is_title_centered?: boolean;
+    is_vertical_bottom?: boolean;
+    is_vertical_centered?: boolean;
+    is_vertical_top?: boolean;
+    onEntered?: () => void;
+    onExited?: () => void;
+    onMount?: () => void;
+    onUnmount?: () => void;
+    portalId?: string;
+    renderTitle?: () => React.ReactNode;
+    should_header_stick_body?: boolean;
+    small?: boolean;
+    title?: string | React.ReactNode;
+    toggleModal?: (e?: React.MouseEvent<HTMLElement>) => void;
+    width?: string;
 };
```

Comment:
```suggestion
type TModal = TModalElement & {
    exit_classname?: string;
    onEntered?: () => void;
    onExited?: () => void;
};
```
@yauheni-kryzhyk-deriv there are only 3 unique props for Modal, and the rest are also passed to its child - ModalElement🙂

Reply:
oh... thanks... this is a big fail from my side :)

Reply:
@maryia-binary, oh... thanks... this is a big fail from my side :)